llvm/unittests/CAS/ActionCacheTest.cpp

```diff
@@ -0,0 +1,55 @@
+//===- ActionCacheTest.cpp ------------------------------------------------===//
+//
+// Part of the LLVM Project, under the Apache License v2.0 with LLVM Exceptions.
+// See https://llvm.org/LICENSE.txt for license information.
+// SPDX-License-Identifier: Apache-2.0 WITH LLVM-exception
+//
+//===----------------------------------------------------------------------===//
+
+#include "CASTestConfig.h"
+#include "llvm/CAS/ActionCache.h"
+#include "llvm/CAS/CASDB.h"
+#include "llvm/Config/llvm-config.h"
+#include "llvm/Support/FileSystem.h"
+#include "llvm/Testing/Support/Error.h"
+#include "llvm/Testing/Support/SupportHelpers.h"
+#include "gtest/gtest.h"
+
+using namespace llvm;
+using namespace llvm::cas;
+
+TEST_P(CASTest, ActionCacheHit) {
+  std::unique_ptr<CASDB> CAS = createCAS();
+  std::unique_ptr<ActionCache> Cache = createActionCache(*CAS);
+
+  Optional<ObjectProxy> ID;
+  ASSERT_THAT_ERROR(CAS->createProxy(None, "1").moveInto(ID), Succeeded());
+  ASSERT_THAT_ERROR(Cache->put(*ID, CAS->getReference(*ID)), Succeeded());
+  Optional<ObjectRef> Result = Cache->get(*ID);
+  ASSERT_TRUE(Result);
+  ASSERT_EQ(*ID, *Result);
+}
+
+TEST_P(CASTest, ActionCacheMiss) {
+  std::unique_ptr<CASDB> CAS = createCAS();
+  std::unique_ptr<ActionCache> Cache = createActionCache(*CAS);
+
+  Optional<ObjectProxy> ID1, ID2;
+  ASSERT_THAT_ERROR(CAS->createProxy(None, "1").moveInto(ID1), Succeeded());
+  ASSERT_THAT_ERROR(CAS->createProxy(None, "2").moveInto(ID2), Succeeded());
+  ASSERT_THAT_ERROR(Cache->put(*ID1, CAS->getReference(*ID2)), Succeeded());
+  // This is a cache miss for looking up a key doesn't exist.
+  ASSERT_FALSE(Cache->get(*ID2));
+}
+
+TEST_P(CASTest, ActionCachePoison) {
```

Comment:
This test is named "poison", but does not test the `poison` method, which seems odd.